com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs

```diff
@@ -48,15 +48,28 @@ internal sealed class iOSNotificationsManager : INotificationsManager {
         [DllImport("__Internal")] private static extern void _notificationsWillDisplayEventPreventDefault(string notificationId);
         [DllImport("__Internal")] private static extern void _notificationsSetClickCallback(ClickListenerDelegate callback);
 
-        public delegate void PermissionListenerDelegate(bool permission);
+        private delegate void PermissionListenerDelegate(bool permission);
         private delegate void WillDisplayListenerDelegate(string notification);
         private delegate void ClickListenerDelegate(string notification, string resultActionId, string resultUrl);
         private delegate void BooleanResponseDelegate(int hashCode, bool response);
 
         public event EventHandler<NotificationWillDisplayEventArgs> ForegroundWillDisplay;
-        public event EventHandler<NotificationClickEventArgs> Clicked;
         public event EventHandler<NotificationPermissionChangedEventArgs> PermissionChanged;
 
+        private bool _clickDelegate;
```

Comment:
I would recommend a more descriptive name here. Something like `clickNativeListenerSet`.
I also recommend a short description about why this boolean is important.